### PR TITLE
✨Basic Cli for exporting top-level CMake file

### PR DESCRIPTION
**Pull request by rosiemiller2**
### Description

Adds a basic Cli for exporting the top-level cmake file.

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_build/_internal/cmake_renderer.py [hunk 1]
```diff
@@ -3,41 +3,25 @@
...
-
-    output_file = export_directory.joinpath("CMakeLists.txt")
-    output_file.write_text(_render_cmakelists_template())
-
-
-def _render_cmakelists_template() -> str:
+def render_cmakelists_template(target_labels: List[str], toolchain_labels: List[str]) -> str:
     """Loads the CMakeLists.txt file template and renders it with the correct details.
 
+    Arguments:
```
Comment (muchzill4) [inline mbed_build/_internal/cmake_renderer.py]: Is it `Arguments` or `Args`? 🤔 Does it matter?

File: mbed_build/mbed_build.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    toolchain_labels = _fetch_toolchain_labels(toolchain_name)
+    return render_cmakelists_template(target_labels, toolchain_labels)
+
+
+def export_cmakelists_file(output_directory: pathlib.Path, file_contents: str) -> None:
+    """Writes out file contents to a CMakeLists.txt file in the output directory.
+
+    Args:
+        output_directory: the destination directory for the exported top-level CMakeLists.txt file
+        file_contents: the contents of the top-level CMakeLists.txt file
```
Comment (muchzill4) [inline mbed_build/mbed_build.py]: NIT: missing `Raises:` - is it worth adding?

File: mbed_build/_internal/mbed_tools/export.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    "-o",
+    "--output_directory",
+    type=click.Path(),
+    required=True,
+    help="Destination for exported top-level CMakeLists.txt file.",
+)
+@click.option(
+    "-t",
+    "--toolchain",
+    type=click.Choice(["ARM5", "ARM6", "GCC", "IAR"]),
```
Comment (madchutney) [inline mbed_build/_internal/mbed_tools/export.py]: ARM C5 will not be supported and IAR is still being discussed but best not to put too much effort into this one.

File: mbed_build/_internal/mbed_tools/export.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    required=True,
+    help="The toolchain you are using to build your app.",
+)
+@click.option("-m", "--mbed_target", required=True, help="A build target for an Mbed-enabled device, eg. K64F")
```
Comment (madchutney) [inline mbed_build/_internal/mbed_tools/export.py]: 👍 nice use of `mbed_target`

File: mbed_build/_internal/mbed_tools/export.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import click
+
+from mbed_build.mbed_build import generate_cmakelists_file, write_cmakelists_file
+
+
+@click.command()
+@click.option(
+    "-o",
+    "--output_directory",
```
Comment (madchutney) [inline mbed_build/_internal/mbed_tools/export.py]: Question, should we use `--output-directory` with a hyphen instead?

File: mbed_build/_internal/templates/CMakeLists.tmpl [hunk 6]
```diff
@@ -1 +1,6 @@
-{{ hello }} {{ world }}
+ToDo: use the correct definitions in the current versions of dev_cmake branches
+Not currently representative of final output
```
Comment (madchutney) [inline mbed_build/_internal/templates/CMakeLists.tmpl]: Do these need to be comment lines?

File: mbed_build/_internal/mbed_tools/export.py (hunk 3 above)
Comment (rosiemiller2) [reply mbed_build/_internal/mbed_tools/export.py]: Thats interesting, so the only two options will be "ARM6" and "GCC"?

Comment (rosiemiller2) [reply mbed_build/_internal/mbed_tools/export.py]: If its decided not to continue with IAR

File: mbed_build/_internal/templates/CMakeLists.tmpl (hunk 6 above)
Comment (rosiemiller2) [reply mbed_build/_internal/templates/CMakeLists.tmpl]: It would probably be better 👍

File: mbed_build/_internal/cmake_renderer.py (hunk 1 above)
Comment (rwalton-arm) [reply mbed_build/_internal/cmake_renderer.py]: Google style guide says `Args`. I'd suggest we just follow their guidelines for docstrings.

Comment (rosiemiller2) [reply mbed_build/_internal/cmake_renderer.py]: 👍 Ill change it

File: mbed_build/exceptions.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+#
+"""Public exceptions raised by the package."""
+from mbed_tools_lib.exceptions import ToolsError
+
+
+class MbedBuildError(ToolsError):
+    """Base public exception for the mbed-devices package."""
+
+
+class NotValidExportOutputDirectory(MbedBuildError):
```
Comment (rwalton-arm) [inline mbed_build/exceptions.py]:
```suggestion
class InvalidExportOutputDirectory(MbedBuildError):
```

**Comment by rosiemiller2** [comment]
> A+! Would review again.
> 
> 1. VERY NITPICK: I know we haven't been consistent with that in the past, but I wonder if `verb` orientated names of files wouldn't be better: `cmake_renderer` -> `render_cmake`, it's similar to how `test_something` sounds.
> 2. VERY NITPICK: I'm not sure what the plan is and usually I'm in favour of tiny files, but doesn't it make sense to group `cmake` things together? Be it under one file or under a module? I.e.:
> 
> ```
> _internal/cmake/render.py
> _internal/cmake/write.py
> ```
> 
> Or `_internal/cmake_file.py`?

I to-ed and fro-ed about this for while but I think you are right and separate files is excessive atm - ccb5ae1.

File: mbed_build/_internal/cmake_renderer.py (hunk 1 above)
Comment (rosiemiller2) [reply mbed_build/_internal/cmake_renderer.py]: 079dca8

File: mbed_build/mbed_build.py (hunk 2 above)
Comment (rosiemiller2) [reply mbed_build/mbed_build.py]: d8d0f4f

File: mbed_build/exceptions.py (hunk 7 above)
Comment (rosiemiller2) [reply mbed_build/exceptions.py]: d8d0f4f

File: mbed_build/_internal/mbed_tools/export.py (hunk 3 above)
Comment (rosiemiller2) [reply mbed_build/_internal/mbed_tools/export.py]: e1fbf9b - I have only included the ones we know we will support for now. We can add IAR back in if its wanted later.

File: mbed_build/_internal/mbed_tools/export.py (hunk 5 above)
Comment (rosiemiller2) [reply mbed_build/_internal/mbed_tools/export.py]: edb1dde

File: mbed_build/_internal/templates/CMakeLists.tmpl (hunk 6 above)
Comment (rosiemiller2) [reply mbed_build/_internal/templates/CMakeLists.tmpl]: a8f33df

**Pull request merged**